app/javascript/src/hyacinth_v1/utils/lazyFontAwesome.js

```diff
@@ -0,0 +1,19 @@
+import React, { Suspense } from 'react';
+import PropTypes from 'prop-types';
+
+const LazyFontAwesomeIcon = (props) => {
+  const { fallback, icon, ...rest } = props;
+  const FontAwesomeIcon = React.lazy(() => import(/* webpackChunkName: 'fontAwesome' */ './fontAwesome'));
+  return <Suspense fallback={fallback}><FontAwesomeIcon icon={icon} { ...rest } /></Suspense>;
+};
+
+LazyFontAwesomeIcon.propTypes = {
+  fallback: PropTypes.object,
+  icon: PropTypes.string.isRequired,
+};
+
+LazyFontAwesomeIcon.defaultProps = {
+  fallback: <span />,
+};
+
+export default LazyFontAwesomeIcon;
```

Comment:
Very creative solution with this module (and lazyJSONInput.js). I like this pattern a lot better than having to use the magic comments in many different places.

Reply:
I'm not gonna lie, I was pretty pleased with this refactor.